src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,80 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String DELIMITER = ",";
+    private static final String LS = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] totals = readAndAggregate(fromFileName);
+        String report = buildReport(totals[0], totals[1]);
+        writeReport(toFileName, report);
+    }
+
+    private int[] readAndAggregate(String fromFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        try (BufferedReader br = Files.newBufferedReader(
+                Paths.get(fromFileName), StandardCharsets.UTF_8)) {
+            String line;
+            int lineNumber = 0;
+            while ((line = br.readLine()) != null) {
+                lineNumber++;
+                String trimmedLine = line.trim();
+                if (trimmedLine.isEmpty()) {
+                    continue;
+                }
+                String[] parts = trimmedLine.split(DELIMITER);
+                if (parts.length != 2) {
+                    continue;
+                }
+                String operation = parts[0].trim();
+                String amountRaw = parts[1].trim();
+                final int amount;
+                try {
+                    amount = Integer.parseInt(amountRaw);
+                } catch (NumberFormatException e) {
+                    throw new RuntimeException("Invalid amount "
+                            + amountRaw + " at line "
+                            + lineNumber + "in file "
+                            + fromFileName, e);
+                }
+                if (SUPPLY.equals(operation)) {
+                    supplyTotal += amount;
+                } else if (BUY.equals(operation)) {
+                    buyTotal += amount;
+                }
```

Unknown operations are silently ignored (only `supply` and `buy` are handled). This violates: "Lines with unknown operations should be treated as invalid and trigger exception per error handling guideline [CHECKLIST ITEM #11]." If `operation` is not `supply` or `buy`, throw a RuntimeException with details (operation, line, file).